src/server/saved_objects/routes/find.js

```diff
@@ -0,0 +1,32 @@
+import Joi from 'joi';
+import { keysToCamelCaseShallow } from '../../../utils/case_conversion';
+
+export const createFindRoute = (prereqs) => ({
+  path: '/api/saved_objects/{type?}',
+  method: 'GET',
+  config: {
+    pre: [prereqs.getSavedObjectsClient],
+    validate: {
+      params: Joi.object().keys({
+        type: Joi.string()
+      }),
+      query: Joi.object().keys({
+        per_page: Joi.number().min(0).default(20),
+        page: Joi.number().min(0).default(1),
+        type: Joi.string(),
+        search: Joi.string().allow('').optional(),
+        search_fields: [Joi.string(), Joi.array().items(Joi.string())],
+        fields: [Joi.string(), Joi.array().items(Joi.string())]
+      })
+    },
+    handler(request, reply) {
+      const options = keysToCamelCaseShallow(request.query);
```

Comment:
> back to snake case when it hits es

It might also be transformed into something entirely different when it hits ES. Setting `search_fields` will transform the query body and `per_page` sets `size'